### DISCUSSION TITLE
How to update hash and querys in url without triggering `route.path` update?

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuejs/router#1687] [Help and Questions] [answered]
When using router, there might be a lot of computed variables and watchers based on `route.path`, meanwhile, a common need for docs is to update url hash while the page is scrolling.

In vuepress2, we use the following lines to achieve it:

https://github.com/vuepress/vuepress-next/blob/fd2e241235fc87c66ce59ba22c8e27521c594fbd/ecosystem/plugin-active-header-links/src/client/composables/useActiveHeaderLinks.ts#L118-L130

However, these codes will trigger the whole update of `route`, and any computed variables and watchers will be recalcuated (or rerun) while scrolling page, which cause performace impact.

So my concern is how can we update hash and query to let `route.hash` and `route.query` being updated in page, without triggering `route.path` update?

**Comment by posva** [comment, marked as answer]
If you watch the path and it doesn't change, the watcher won't trigger:

```ts
watch(() => route.path, () => {
  // ...
})
```
